pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -120,7 +120,7 @@ public static SqlNodeAndOptions compileToSqlNodeAndOptions(String sql)
       sqlNodeAndOptions.setParseTimeNs(System.nanoTime() - parseStartTimeNs);
       return sqlNodeAndOptions;
     } catch (Throwable e) {
-      throw new SqlCompilationException("Caught exception while parsing query: " + sql, e);
+      throw new SqlCompilationException("Caught exception while parsing query: " + sql + ": " + e.getMessage(), e);
```

Comment:
This is mandatory, otherwise the message doesn't include the actual compilation issue